crates/runc/src/lib.rs

```diff
@@ -0,0 +1,980 @@
+        ];
+        if let Some(opts) = opts {
+            args.append(&mut opts.args()?);
+        }
+        args.push(id.to_string());
+        let mut cmd = self.command(&args)?;
+        match opts {
+            Some(CreateOpts { io: Some(_io), .. }) => {
+                _io.set_tk(&mut cmd).map_err(Error::UnavailableIO)?;
+                let (tx, rx) = tokio::sync::oneshot::channel::<Exit>();
+                let start = MONITOR.start(cmd, tx);
+                let wait = MONITOR.wait(rx);
+                let (
+                    Output {
+                        status,
+                        stdout,
+                        stderr,
+                    },
+                    _,
+                ) = tokio::try_join!(start, wait).map_err(Error::InvalidCommand)?;
+                _io.close_after_start();
+
+                let stdout = String::from_utf8(stdout).unwrap();
+                let stderr = String::from_utf8(stderr).unwrap();
+                if !status.success() {
+                    return Err(Error::CommandFailed {
+                        status,
+                        stdout,
+                        stderr,
+                    });
+                }
+            }
+            _ => {
+                let _ = self.launch(cmd, true).await?;
+            }
+        }
+        Ok(())
+    }
+
+    /// Delete a container
+    pub async fn delete(&self, id: &str, opts: Option<&DeleteOpts>) -> Result<()> {
+        let mut args = vec!["delete".to_string()];
+        if let Some(opts) = opts {
+            args.append(&mut opts.args());
+        }
+        args.push(id.to_string());
+        let _ = self.launch(self.command(&args)?, true).await?;
+        Ok(())
+    }
+
+    /// Return an event stream of container notifications
+    pub async fn events(&self, _id: &str, _interval: &Duration) -> Result<()> {
+        Err(Error::Unimplemented("events".to_string()))
+    }
+
+    /// Execute an additional process inside the container
+    pub async fn exec(&self, _id: &str, _spec: &Process, _opts: Option<&ExecOpts>) -> Result<()> {
+        Err(Error::Unimplemented("exec".to_string()))
+    }
+
+    /// Send the specified signal to processes inside the container
+    pub async fn kill(&self, id: &str, sig: u32, opts: Option<&KillOpts>) -> Result<()> {
+        let mut args = vec!["kill".to_string()];
+        if let Some(opts) = opts {
+            args.append(&mut opts.args());
+        }
+        args.push(id.to_string());
+        args.push(sig.to_string());
+        let _ = self.launch(self.command(&args)?, true).await?;
+        Ok(())
+    }
+
+    /// List all containers associated with this runc instance
+    pub async fn list(&self) -> Result<Vec<Container>> {
+        let args = ["list".to_string(), "--format-json".to_string()];
+        let res = self.launch(self.command(&args)?, true).await?;
+        let output = res.output.trim();
+        // Ugly hack to work around golang
```

Comment:
I want to know more about this :)

Reply:
Ah, this is for just handling `null` json. But I think this way (with comparering output and "null") is not concise...
Is it comprehensive if I change it like below?
```rust
Ok(serde_json::from_str::<Option<Vec<usize>>>(output).map_err(/*omit*/)?.unwrap_or_default())
```